src/melodyPlayer/Keyboard.jsx

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable functional/immutable-data */
+import * as React from 'react';
+import styles from './keyboard.module.css';
+
+export function Keyboard() {
+  const totalKeyss = [0, 1, 2, 3];
+  const keyNames = [
+    'C',
+    'C#/Db',
+    'D',
+    'D#/Eb',
+    'E',
+    'F',
+    'F#/Gb',
+    'G',
+    'G#/Ab',
+    'A',
+    'A#/Bb',
+    'B',
+  ];
+  const keyboard = document.getElementById('keyboard');
+  const totalKeys = 14;
+  const startingOctave = 2;
+
+  const allBlackKeysArray = Array.from(
+    document.getElementsByClassName('blackkey')
+  );
+
+  function addKeys() {
+    const keysPerOctave = keyNames.length;
+    const i = 0;
+    const keyCount = 0;
+    const octaveCount = startingOctave; //start at the second octave;
+    while (i < totalKeys) {
+      addWhiteKey(keyCount, octaveCount);
+      keyCount++;
+
+      if (keyCount < keysPerOctave) {
+        if (blackKeyCheck(keyCount)) {
+          addBlackKey(keyCount, octaveCount);
+          keyCount++;
+        }
+      } else {
+        keyCount = 0;
+        octaveCount++;
+      }
+
+      i++;
+    }
+
+    const totalKeyCount = keyboard.childElementCount;
+    document.getElementById(
+      'description'
+    ).innerHTML = `Below should be a keyboard with ${totalKeyCount} keys.`;
+  }
+
+  function addWhiteKey(keyCount, octaveCount) {
+    const newDiv = document.createElement('div');
+    newDiv.classList.add(styles.whitekey);
+    console.log(newDiv)
+    newDiv.setAttribute('key-name', keyNames[keyCount]);
+    newDiv.setAttribute('octave', octaveCount);
+    keyboard.appendChild(newDiv);
+  }
+
+  function addBlackKey(keyCount, octaveCount) {
+    const newDiv = document.createElement('div');
+    newDiv.classList.add(styles.blackkey);
+    newDiv.setAttribute('key-name', keyNames[keyCount]);
+    newDiv.setAttribute('octave', octaveCount);
+    keyboard.appendChild(newDiv);
+    sizeBlackKey(newDiv);
+  }
+
+  function blackKeyCheck(keyCount) {
+    const whiteKeysNeedingBlackKeys = [1, 3, 6, 8, 10];
+    return whiteKeysNeedingBlackKeys.includes(keyCount);
+  }
+
+  function resizeAllBlackKeys() {
+    allBlackKeysArray.forEach(sizeBlackKey);
+  }
+
+  function sizeBlackKey(blackKey) {
+    const whiteKeySibling = blackKey.previousElementSibling;
+    const whiteKeySiblingCoords = getPos(whiteKeySibling);
+    blackKey.style.width = `${whiteKeySibling.offsetWidth / 2}px`;
+    blackKey.style.left = `${
+      whiteKeySiblingCoords.right - blackKey.offsetWidth / 2
+    }px`;
+  }
+
+  function getPos(el) {
+    return el.getBoundingClientRect();
+  }
+
+  return (
+    <>
+      <div id="description"></div>
+      <div id="keyboard"></div>
+      <script>
+        {(document.body.onload = addKeys)}
+        {window.addEventListener('load', resizeAllBlackKeys)}
+        {window.addEventListener('resize', resizeAllBlackKeys)}
+      </script>
```

Comment:
@dancingshell looks like this doesn't do anything - mind pointing me in the right direction?

Reply:
ha this is funny. basically the script tag is a way to do javascript in html. but now the whole component is where you can do javascript code. so you can move these listeners higher in the component. also this will likely be where you want to start using hooks. the hook `useEffect` will let you put code in the component that you only want to run at specific times. this is good for event listeners where you only want to add one the first time the component mounts